src/wallet/wallet_rpc_server.cpp

```diff
@@ -1424,6 +1427,54 @@ namespace tools
     return true;
   }
   //------------------------------------------------------------------------------------------------------------------------------
+  bool wallet_rpc_server::on_start_mining(const wallet_rpc::COMMAND_RPC_START_MINING::request& req, wallet_rpc::COMMAND_RPC_START_MINING::response& res, epee::json_rpc::error& er)
+  {
+    if (!m_trusted_daemon)
+    {
+      er.code = WALLET_RPC_ERROR_CODE_UNKNOWN_ERROR;
+      er.message = "This command requires a trusted daemon.";
+      return false;
+    }
+
+    size_t max_mining_threads_count = (std::max)(tools::get_max_concurrency(), static_cast<unsigned>(2));
+    if (req.threads_count < 1 || max_mining_threads_count < req.threads_count)
+    {
+      er.code = WALLET_RPC_ERROR_CODE_UNKNOWN_ERROR;
+      er.message = "The specified number of threads is inappropriate.";
+      return false;
+    }
+
+    cryptonote::COMMAND_RPC_START_MINING::request daemon_req = AUTO_VAL_INIT(daemon_req); 
+    daemon_req.miner_address = m_wallet.get_account().get_public_address_str(m_wallet.testnet());
+    daemon_req.threads_count        = req.threads_count;
+    daemon_req.do_background_mining = req.do_background_mining;
+    daemon_req.ignore_battery       = req.ignore_battery;
+
+    cryptonote::COMMAND_RPC_START_MINING::response daemon_res;
+    bool r = net_utils::invoke_http_json("/start_mining", daemon_req, daemon_res, m_http_client);
+    if (!r || daemon_res.status != CORE_RPC_STATUS_OK)
+    {
+      er.code = WALLET_RPC_ERROR_CODE_UNKNOWN_ERROR;
```

Comment:
`WALLET_RPC_ERROR_CODE_UNKNOWN_ERROR` is used pretty much everywhere in `wallet_rpc_server.cpp` where a call to a `wallet2` member function can throw. The slight difference with this code is that it directly calls a daemon RPC.